pkgs/by-name/fa/fastcap/package.nix

```
@@ -0,0 +1,88 @@
+{ stdenv
+, fetchzip
+, ghostscript
+, lib
+, texlive
+}:
+
+stdenv.mkDerivation rec {
+  pname = "fastcap";
+  version = "2.0-18Sep92";
+
+  src = fetchzip {
+    url = "http://www.rle.mit.edu/cpg/codes/fastcap-${version}.tgz";
```

Consider using the `finalAttrs` pattern. It helps making the derivation easier to override.

```suggestion
stdenv.mkDerivation (finalAttrs: {
  pname = "fastcap";
  version = "2.0-18Sep92";

  src = fetchzip {
    url = "http://www.rle.mit.edu/cpg/codes/fastcap-${finalAttrs.version}.tgz";
```

---

I'm not particularly keen to introduce ad-hoc tricks for two reasons: (i) only a few packages do this and having differences in packages requires different solutions to the same problem, depending on the package. (ii) A uniform change is better achieved, e.g, within `mkDerivation`.

---

1. It is not ad-hoc. It is precisely an application of Y Combinator.
2. The `finalAttrs` design pattern can be applied in any instance of `mkDerivation`, therefore it is very "uniform".

That being said, `finalAttrs` and `rec` are both allowed in Nixpkgs - besides personal opinions and technical merits of each approach.

---

If both patterns are allowed, I'll stick with `rec`.